lab-justin/lib/client-pool.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+var uuid = require('node-uuid');
+
+function registerClient(socket, clientPool) {
+  socket.wack = {};
```

Comment:
You're adding a socket.wack object in two different places that overwrite each other.